File: archaic/src/main/scala/resonantinduction/archaic/process/RenderMillstone.java
package resonantinduction.archaic.process;

import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import resonant.lib.render.RenderItemOverlayUtility;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderMillstone extends TileEntitySpecialRenderer
{
	private final RenderBlocks renderBlocks = new RenderBlocks();

	@Override
	public void renderTileEntityAt(TileEntity tileEntity, double x, double y, double z, float var8)
	{
		if (tileEntity instanceof TileMillstone)
		{
			TileMillstone tile = (TileMillstone) tileEntity;
			RenderItemOverlayUtility.renderItemOnSides(tileEntity, tile.getStackInSlot(0), x, y, z, "");
		}
	}
}

File: src/main/scala/resonantinduction/archaic/process/RenderMillstone.scala
package resonantinduction.archaic.process

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import resonant.lib.render.RenderItemOverlayUtility

@SideOnly(Side.CLIENT)
class RenderMillstone extends TileEntitySpecialRenderer
{
    private final val renderBlocks: RenderBlocks = new RenderBlocks

    def renderTileEntityAt(tileEntity: TileEntity, x: Double, y: Double, z: Double, var8: Float)
    {
        if (tileEntity.isInstanceOf[TileMillstone])
        {
            val tile: TileMillstone = tileEntity.asInstanceOf[TileMillstone]
            RenderItemOverlayUtility.renderItemOnSides(tileEntity, tile.getStackInSlot(0), x, y, z, "")
        }
    }

}